oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/hierarchy/service/ServiceHierarchyRelationTraffic.java

```diff
@@ -0,0 +1,163 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.core.hierarchy.service;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.skywalking.oap.server.core.analysis.Layer;
+import org.apache.skywalking.oap.server.core.analysis.MetricsExtension;
+import org.apache.skywalking.oap.server.core.analysis.Stream;
+import org.apache.skywalking.oap.server.core.analysis.metrics.Metrics;
+import org.apache.skywalking.oap.server.core.analysis.worker.MetricsStreamProcessor;
+import org.apache.skywalking.oap.server.core.remote.grpc.proto.RemoteData;
+import org.apache.skywalking.oap.server.core.source.DefaultScopeDefine;
+import org.apache.skywalking.oap.server.core.storage.StorageID;
+import org.apache.skywalking.oap.server.core.storage.annotation.BanyanDB;
+import org.apache.skywalking.oap.server.core.storage.annotation.Column;
+import org.apache.skywalking.oap.server.core.storage.type.Convert2Entity;
+import org.apache.skywalking.oap.server.core.storage.type.Convert2Storage;
+import org.apache.skywalking.oap.server.core.storage.type.StorageBuilder;
+
+@Stream(name = ServiceHierarchyRelationTraffic.INDEX_NAME, scopeId = DefaultScopeDefine.SERVICE_HIERARCHY_RELATION,
+    builder = ServiceHierarchyRelationTraffic.Builder.class, processor = MetricsStreamProcessor.class)
+@MetricsExtension(supportDownSampling = false, supportUpdate = false)
+@EqualsAndHashCode(of = {
+    "entityId"
+}, callSuper = true)
+public class ServiceHierarchyRelationTraffic extends Metrics {
+    public static final String INDEX_NAME = "service_hierarchy_relation";
+    public static final String SERVICE_ID = "service_id";
+    public static final String SERVICE_LAYER = "service_layer";
+    public static final String RELATED_SERVICE_ID = "related_service_id";
+    public static final String RELATED_SERVICE_LAYER = "related_service_layer";
+
+    @Setter
+    @Getter
+    @Column(name = ENTITY_ID, length = 512)
+    @BanyanDB.SeriesID(index = 0)
+    private String entityId;
+
+    @Setter
+    @Getter
+    @Column(name = SERVICE_ID, length = 250)
+    private String serviceId;
+
+    @Setter
+    @Getter
+    @Column(name = SERVICE_LAYER)
+    private Layer servicelayer = Layer.UNDEFINED;
+
+    @Setter
+    @Getter
+    @Column(name = RELATED_SERVICE_ID, length = 250)
+    private String relatedServiceId;
+
+    @Setter
+    @Getter
+    @Column(name = RELATED_SERVICE_LAYER)
+    private Layer relatedServiceLayer = Layer.UNDEFINED;
```

Comment:
I think for `ServiceHierarchyRelationTraffic`, we could skip all these fields, as ID of the relationship should be able to be decoded, right? You don't add the decoding in IDManager, I believe it is possible.

By that, and service hierarchy would be loaded in the memory, so, we should be able to optimize the storage?